app/views/software_records/_upgrade_history.html.erb

```diff
@@ -1 +1,49 @@
-Upgrade history
+Upgrade history (Completed Change Request)
+
+<% if @change_request_counts != 0 %>
+  <div class="container animated fadeInUp" style="margin-bottom: 10rem;">
+    <div class="row">
+      <table class="table text-center">
+        <thead>
+          <tr>
+            <th class="text-left" style="padding-left: 50px">Date Submitted</th>
+            <th class="text-left" style="padding-left: 50px">Change Requests</th>
+            <th class="text-left" style="padding-left: 50px">Version</th>
+            <th class="text-left" style="padding-left: 50px">Tech Lead</th>
+
+            <% if current_user.role.to_s == "viewer" %>
+              <th>Actions</th>
+            <% elsif current_user.role.to_s == "owner" %>
+              <th colspan="2">Actions</th>
+            <% else %>
+              <th colspan="3">Actions</th>
+            <% end %>
+          </tr>
+        </thead>
+
+        <tbody>
+          <% software_records_upgrade_hash(@software_record.id).each do |change_request| %>
+            <tr>
+              <td class="text-left" style="padding-left: 50px"><%= change_request.change_submitted_date %></td>
+              <td class="text-left" style="padding-left: 50px"><%= change_request.change_title %></td>
+              <td class="text-left" style="padding-left: 50px"><%= find_software_version(change_request.software_record_id) %></td>
+              <td class="text-left" width="250px" style="padding-left: 50px"><% find_tech_leads(change_request.software_record_id).each do |a| %><li><%= a %></li><% end %></td>
+              <% if current_user.role.to_s == "viewer" %>
+                <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
+              <% elsif current_user.role.to_s == "owner" %>
+                <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
+                <td><%= link_to 'Edit', edit_change_request_path(change_requests), { :class => "btn btn-primary" } %></td>
```

Comment:
```
ActionView::Template::Error (undefined local variable or method `change_requests' for #<ActionView::Base:0x0000000000e9e8>
Did you mean?  change_request
               change_requests_url
               change_request_url):
    32:                 <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
    33:               <% elsif current_user.role.to_s == "owner" %>
    34:                 <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
    35:                 <td><%= link_to 'Edit', edit_change_request_path(change_requests), { :class => "btn btn-primary" } %></td>
    36:               <% else %>
    37:                 <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
    38:                 <td><%= link_to 'Edit', edit_change_request_path(change_request), { :class => "btn btn-primary" } %></td>
  
app/views/software_records/_upgrade_history.html.erb:35
app/views/software_records/_upgrade_history.html.erb:25
app/views/software_records/show.html.erb:59
```

Reply:
little typo: argument "change_requests" in `edit_change_request_path` helper should be singular "change_request"